ts/client/src/accounts/mangoAccount.ts

```diff
@@ -231,9 +231,24 @@ export class MangoAccount {
    * @param bank
    * @returns native balance for a token, is signed
    */
-  public getTokenBalance(bank: Bank): I80F48 {
+  public getTokenBalance(group: Group, bank: Bank): I80F48 {
     const tp = this.getToken(bank.tokenIndex);
-    return tp ? tp.balance(bank) : ZERO_I80F48();
+    if (tp) {
+      const bal = tp.balance(bank);
+      for (const serum3Market of Array.from(
+        group.serum3MarketsMapByMarketIndex.values(),
+      )) {
+        const oo = this.serum3OosMapByMarketIndex.get(serum3Market.marketIndex);
```

Comment:
mh. if we need the group only to get all the serum3 market indexes, why don't we just iterate over the key, value of serum3OosMapByMarketIndex?

Reply:
we also need to know the base token index and the quote token index